R/finalize_plot.R

```diff
@@ -261,6 +261,20 @@ finalize_plot <- function(plot = NULL,
     new = list(size = ggplot_size_conversion(consts$lwd_plotline))
     )
 
+  # Font defaults for annotations and labels on the graph
+  default_fontfamily <- ggplot2::theme_get()$text$family
+  default_fontface <- ggplot2::theme_get()$text$face
+  default_fontcolor <- ggplot2::theme_get()$text$colour
+  default_fontsize <- ggplot2::theme_get()$text$size
+
+  ggplot2::update_geom_defaults(
+    geom = "text",
+    new = list(family = cmapplot_globals$font$label$family,
+               face = cmapplot_globals$font$label$face,
+               size = cmapplot_globals$font$label$size,
+               colour = cmapplot_globals$colors$blackish)
```

Comment:
Can this be extended to apply to `geom_text_lastonly`?

Reply:
I think so? Some of this may no longer be necessary after we update to the complete theme, as I think that solves some of the inheritance problems...so we will have to redo some testing before doing anything with this branch.

Reply:
I believe we can set `geom_text_lastonly` defaults in the `default_aes` section of the `GeomTextLast` ggproto object created in geom_text_lastonly.R. I haven't experimented with whether `update_geom_defaults()` works with custom ggproto objects, although I would assume it does not.

Reply:
@matthewstern if I remember our earlier conversations correctly, I had proposed modifying the function to use the CMAP font spec by default, but we decided not to implement that because it would look weird when `theme_cmap` was not also being used.

Reply:
@matthewstern @dlcomeaux one more note on this: if we export the `GeomTextLast` ggproto object (as ggplot2 does for its own ggproto objects) then we should be able to use `update_geom_defaults("TextLast", ...)` on it.

Reply:
In commit 09584be I have exported all our proto objects and created dummy documentation for them. I can confirm that `update_geom_defaults` does work on our custom geoms. 

Test code:
```
g <- ggplot(mtcars, aes(x = mpg, y = disp, label = rownames(mtcars))) + 
  geom_text() +
  geom_text_lastonly()

g # all text black

update_geom_defaults("Text", list(color = "blue"))
update_geom_defaults("TextLast", list(color = "red"))

g # text now blue and red
```